src/sentry/auth/providers/saml2/provider.py

```diff
@@ -290,7 +290,7 @@ def build_identity(self, state):
             error_msg_keys = ", ".join(repr(key) for key in sorted(raw_attributes.keys()))
             raise IdentityNotValid(
                 _(
-                    f"Failed to map SAML attributes. Assertion returned the following attribute keys: {error_msg_keys}"
```

Comment:
Actually - now i'm confused.

Is this error trying to tell us that "we received the following attribute keys from the identity provider"?
Or is this error trying to tell us "the assertion failed on the following attribute keys"?

this PR is to address the latter.

@evanpurkhiser would you know?

Reply:
If it's the former, then this error message doesn't tell us anything about which mapping on our side was incorrect 😮‍💨

Reply:
Double check me, but I believe this error happens when the list of possible identifiers returned by the saml assertion does not include one or more of the user provided identifiers.

This message is basically saying “you put something  in wrong, here’s the identifiers the saml server is telling us you can use”

Reply:
Oh 🤔 

So it _is_ the former?
Ok that checks out then 🤔 

will update this PR